pkg/operator/controller/machineconfig/controller.go

```diff
@@ -0,0 +1,360 @@
+			r.Log.Error(err, "failed to enable kubelet profiling")
+			errs = fmt.Errorf("%w: %s", errs, err.Error())
+		}
+		if created {
+			r.EventRecorder.Eventf(r.CtrlConfig, corev1.EventTypeNormal, "CreateConfig", "successfully created kubelet config")
+			r.PrevSyncChange["kubelet"] = PrevSyncData{
+				action: "created",
+				config: *kubeletmc,
+			}
+		}
+	} else {
+		deleted, err := r.ensureKubeletProfConfigNotExists(ctx)
+		if err != nil {
+			errored = true
+			r.Log.Error(err, "failed to disable kubelet profiling")
+			errs = fmt.Errorf("%w: %s", errs, err.Error())
+		}
+		if deleted {
+			r.EventRecorder.Eventf(r.CtrlConfig, corev1.EventTypeNormal, "DeleteConfig", "successfully deleted kubelet config")
+			r.PrevSyncChange["kubelet"] = PrevSyncData{
+				action: "deleted",
+			}
+		}
+	}
+
+	if errored {
+		return errs
+	}
+	return nil
+}
+
+// checkMCPUpdateStatus is for reconciling update status of all machines in profiling MCP
+func (r *MachineconfigReconciler) checkMCPUpdateStatus(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	mcp := &mcv1.MachineConfigPool{}
+	key := types.NamespacedName{Namespace: r.CtrlConfig.Namespace, Name: ProfilingMCPName}
+	if err := r.Client.Get(ctx, key, mcp); err != nil {
+		r.Log.Error(err, "failed to fetch profiling MCP: %v", err)
+		return ctrl.Result{}, err
+	}
+
+	if mcv1.IsMachineConfigPoolConditionTrue(mcp.Status.Conditions, mcv1.MachineConfigPoolUpdating) &&
+		r.CtrlConfig.Status.UpdateStatus.InProgress == "false" {
+		r.EventRecorder.Eventf(r.CtrlConfig, corev1.EventTypeNormal, "ConfigUpdate", "config update still under progress")
+		r.Log.Info("config update under progress")
+		r.CtrlConfig.Status.UpdateStatus.InProgress = corev1.ConditionTrue
+		return ctrl.Result{RequeueAfter: 15 * time.Second}, nil
+	}
+
+	if mcv1.IsMachineConfigPoolConditionTrue(mcp.Status.Conditions, mcv1.MachineConfigPoolUpdated) &&
+		mcp.Status.UpdatedMachineCount == mcp.Status.MachineCount {
+		r.EventRecorder.Eventf(r.CtrlConfig, corev1.EventTypeNormal, "ConfigUpdate", "config update completed on all machines")
+		r.Log.Info("config update completed on all machines")
+		r.CtrlConfig.Status.UpdateStatus.InProgress = "false"
+	} else {
+		if mcp.Status.DegradedMachineCount != 0 {
+			r.EventRecorder.Eventf(r.CtrlConfig, corev1.EventTypeWarning, "ConfigUpdate", "%s MCP has %d machines in degraded state",
+				ProfilingMCPName, mcp.Status.DegradedMachineCount)
+
+			if err := r.revertPrevSyncChanges(ctx); err != nil {
+				return ctrl.Result{RequeueAfter: 15 * time.Second},
+					fmt.Errorf("failed to revert changes to recover degraded machines, will reconcile in 15s")
+			}
+			return ctrl.Result{RequeueAfter: 15 * time.Second},
+				fmt.Errorf("%d machines are in degraded state, will reconcile in 15s", mcp.Status.DegradedMachineCount)
+		}
+		r.Log.Info("waiting for update to finish on all machines", "MachineConfigPool", ProfilingMCPName)
+		return ctrl.Result{RequeueAfter: 15 * time.Second}, nil
```

Comment:
Here too, let's wait longer.

Reply:
comment incorporated in [6e77c7b](https://github.com/openshift/node-observability-operator/pull/16/commits/6e77c7b3116158c01e2e14500d27c633e7997885)
Increased to 3 minutes